packages/@tinacms/app/appFiles/src/preview.tsx

```diff
@@ -97,6 +97,12 @@ const QueryMachine = (props: {
 
   React.useEffect(() => {
     if (props.iframeRef.current) {
+      window.addEventListener('message', (event: MessageEvent<PostMessage>) => {
+        if (event?.data?.type === 'isEditMode') {
+          console.log('sending edit mode')
+          send({ type: 'EDIT_MODE' })
```

Comment:
Instead of sending this to the machine, why not respond back immediately?

```suggestion
          props.iframe?.contentWindow?.postMessage({
            type: 'tina:editMode',
          })
```

Reply:
That makes more sense to me!

Reply:
Fixed